src/Symfony/Bundle/SecurityBundle/SecurityUserValueResolver.php

```diff
@@ -59,4 +59,14 @@ public function resolve(Request $request, ArgumentMetadata $argument)
     {
         yield $this->tokenStorage->getToken()->getUser();
     }
+
+    /**
+     * @param string $type
+     *
+     * @return bool
+     */
+    private function implementsCorrectInterface($type)
+    {
+        return UserInterface::class === $type || array_key_exists(UserInterface::class, class_implements($type));
```

Comment:
Let's inline this logic in the calling condition

Reply:
isn't this what `is_a()` or `is_subclass_of()` does?

Reply:
Didn't know 😳thanks!